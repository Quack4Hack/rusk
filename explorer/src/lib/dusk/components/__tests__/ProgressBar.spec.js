import { afterEach, describe, expect, it } from "vitest";
import { cleanup, render } from "@testing-library/svelte";

import { ProgressBar } from "..";

describe("ProgressBar", () => {
  const baseProps = {
    ariaLabel: "Loading",
    currentPercentage: 0,
  };
  const baseOptions = {
    props: baseProps,
    target: document.body,
  };

  afterEach(cleanup);

  it("renders the `ProgressBar` component with no current percentage set", () => {
    const { container } = render(ProgressBar, {
      ...baseOptions,
      props: { ariaLabel: baseProps.ariaLabel },
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the `ProgressBar` component with current percentage set as zero", () => {
    const { container } = render(ProgressBar, baseOptions);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("re-renders the `ProgressBar` component when the current percentage property changes", async () => {
    const { container, rerender } = render(ProgressBar, baseOptions);

    expect(container.firstChild).toMatchSnapshot();

    await rerender({ currentPercentage: 50 });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should pass additional class names to the rendered element", () => {
    const { container } = render(ProgressBar, {
      ...baseOptions,
      props: { ...baseProps, className: "foo bar" },
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
