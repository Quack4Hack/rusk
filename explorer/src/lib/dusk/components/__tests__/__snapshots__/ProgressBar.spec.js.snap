// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProgressBar > re-renders the \`ProgressBar\` component when the current percentage property changes 1`] = `
<div
  aria-label="Loading"
  class="dusk-progress-bar"
  role="progressbar"
>
  <div
    class="dusk-progress-bar__filler"
    style="width: 0%"
  />
</div>
`;

exports[`ProgressBar > re-renders the \`ProgressBar\` component when the current percentage property changes 2`] = `
<div
  aria-label="Loading"
  class="dusk-progress-bar"
  role="progressbar"
>
  <div
    class="dusk-progress-bar__filler"
    style="width: 50%"
  />
</div>
`;

exports[`ProgressBar > renders the \`ProgressBar\` component with current percentage set as zero 1`] = `
<div
  aria-label="Loading"
  class="dusk-progress-bar"
  role="progressbar"
>
  <div
    class="dusk-progress-bar__filler"
    style="width: 0%"
  />
</div>
`;

exports[`ProgressBar > renders the \`ProgressBar\` component with no current percentage set 1`] = `
<div
  aria-label="Loading"
  class="dusk-progress-bar"
  role="progressbar"
>
  <div
    class="dusk-progress-bar__filler dusk-progress-bar__filler--undetermined"
  />
</div>
`;

exports[`ProgressBar > should pass additional class names to the rendered element 1`] = `
<div
  aria-label="Loading"
  class="dusk-progress-bar foo bar"
  role="progressbar"
>
  <div
    class="dusk-progress-bar__filler"
    style="width: 0%"
  />
</div>
`;
